sympy/polys/rootoftools.py

```diff
@@ -774,13 +774,58 @@ def _get_roots(cls, method, poly, radicals):
             if x.name not in free_names:
                 poly = poly.xreplace({d: x})
                 break
+
+        # compute rational lift if real algebraic
+        has_alg_coeff = any(c.is_algebraic and not c.is_rational
+                            for c in poly.all_coeffs())
+        has_alg_coeff = has_alg_coeff and all(c.is_real
+                            for c in poly.all_coeffs())
+
+        if has_alg_coeff:
+            poly_original = poly
+            poly = Poly(poly.expr, poly.gen, extension=True)
+            poly = poly.lift()
+
         coeff, poly = cls._preprocess_roots(poly)
         roots = []
 
         for root in getattr(cls, method)(poly):
             roots.append(coeff*cls._postprocess_root(root, radicals))
+
+        # filter out extraneous roots if introduced from lifting
+        if has_alg_coeff:
+            if method == "_real_roots":
+                roots = cls._numerically_filter_roots(poly_original, roots, real=True)
+            elif method == "_all_roots":
+                roots = cls._numerically_filter_roots(poly_original, roots, real=False)
+
         return roots
 
+    @classmethod
+    def _numerically_filter_roots(cls, poly_alg, candidates, real):
+        # to get proper root counts
+        if poly_alg.domain not in (QQ, ZZ):
+            poly_alg = Poly(poly_alg.expr, poly_alg.gen, extension=True)
+
+        if real:
+            num_roots = poly_alg.count_roots()
+        else:
+            num_roots = poly_alg.degree()
+
+        prec = 10
+        # compare len(set()) bc expected behavior is for multiple roots
+        while len(set(candidates)) > num_roots:
+            candidates_filtered = []
+            for c in candidates:
+                r_f = poly_alg(c).evalf(prec, maxn=2*prec)
+                if abs(r_f)._prec < 2: candidates_filtered.append(c)
+            prec *= 2
+            candidates = candidates_filtered
```

Comment:
If there are multiple roots then the inner loop should not repeat the calculation for them.